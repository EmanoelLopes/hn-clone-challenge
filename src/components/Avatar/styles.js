import styled from 'styled-components';

export const Picture = styled.img`
  border: 1px solid #ddd;
  border-radius: 4px;
  margin: 0 0 0 1rem;
  max-width: 42px;
`;