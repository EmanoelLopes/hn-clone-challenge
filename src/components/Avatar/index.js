import React from 'react';
import PropTypes from 'prop-types';
import { Picture } from './styles';

function Avatar ({ source, altText, hasBadge, children }) {
  return (
    <div className="avatar">
      { (hasBadge) ? children : null }
      <Picture src={source} alt={altText}/>
    </div>
  );
};

Avatar.propTypes = {
  source: PropTypes.string,
  altText: PropTypes.string,
  hasBadge: PropTypes.bool,
  children: PropTypes.object
};

export default Avatar;