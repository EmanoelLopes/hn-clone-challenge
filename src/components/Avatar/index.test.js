import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Avatar from '../Avatar';

configure({ adapter: new Adapter() });

const props = {
  source: '',
  altText: '',
  hasBadge: true,
  children: {}
};

describe('<Avatar />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Avatar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

