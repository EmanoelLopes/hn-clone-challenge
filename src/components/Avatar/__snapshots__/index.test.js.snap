// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Avatar
    altText=""
    hasBadge={true}
    source=""
  >
    Object {}
  </Avatar>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Object {},
        <styled.img
          alt=""
          src=""
        />,
      ],
      "className": "avatar",
    },
    "ref": null,
    "rendered": Array [
      Object {},
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "",
          "src": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Object {},
          <styled.img
            alt=""
            src=""
          />,
        ],
        "className": "avatar",
      },
      "ref": null,
      "rendered": Array [
        Object {},
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "",
            "src": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
